Feature/success failure

cat-toreda opened the pull request:
created a method to automatically assign success/failure due to errors or data

catwhitmer committed:
added a method to check for success and failure due to errors or data

catwhitmer committed:
added PR review changes

mbrich commented on src/fate.ts:
```diff
@@ -232,6 +231,24 @@ export class Fate<T = unknown> {
+			return this;
+		}
+
+		if (this.data) {
```
I'm not sure if there's a subtle difference between `this.setSuccess(true)` on line 242 and `this.success(true)` on line 247, but i think we can probably remove `if (this.data) {...` and just use the call below.

mbrich commented on src/fate.ts:
```diff
@@ -212,7 +212,6 @@ export class Fate<T = unknown> {
-		this.success(false);
 		this.done(true);
 
 		return this;
```
Is `return this` called twice here, or is my editor showing a duplicate? If `return this` appears twice we can remove one as the second and everything below it will be unreachable code.

mbrich commented on src/fate.ts:
```diff
@@ -10,7 +10,7 @@ export class Fate<T = unknown> {
 	public readonly errorLog: FateObject['errorLog'];
 	public readonly messageLog: FateObject['messageLog'];
-	public readonly errorThreshold: FateObject['errorThreshold'];
+	public errorThreshold: FateObject['errorThreshold'];
```
We can make this property `readonly` and change the type to `UInt` to get the type and value guarantee of `strong-types`.

catwhitmer committed:
added PR changes

catwhitmer committed:
changed the errorThreshold type

SaxanLoster committed:
Fixes for tests.

Infinity is not a valid input for strong UInt, so we can use 999 instead

To set the errorThreshold to 0, we casted errotThreshold as any which
made it not show errors in vsc. With strong types that is no longer
needed.

mbrich committed:
Merge branch 'develop' into feature/success-failure
